pkg/tracking/store/sql/experiments.go

```diff
@@ -252,3 +253,63 @@ func (s TrackingSQLStore) GetExperimentByName(
 
 	return experiment.ToEntity(), nil
 }
+
+func (s TrackingSQLStore) SetExperimentTag(
+	ctx context.Context, experimentID, key, value string,
+) *contract.Error {
+	experiment, err := s.GetExperiment(ctx, experimentID)
+	if err != nil {
+		return err
+	}
+
+	// check experiment is active
+	if models.LifecycleStage(experiment.LifecycleStage) != models.LifecycleStageActive {
```

Comment:
Pleas extract this function like `checkRunIsActive`.
This will probably be useful later on.
The Python code probably has this already.